Makefile

```
@@ -0,0 +1,54 @@
+SHELL := /bin/bash
+
+.PHONY : help init test lint nag release clean
+.DEFAULT: help
+
+VENV_NAME ?= venv
+PYTHON ?= $(VENV_NAME)/bin/python
+
+help:
+	@echo "help	get the full command list"
+	@echo "init	create VirtualEnv and install libraries"
+	@echo "test	run pre-commit checks"
+	@echo "lint	GitHub actions cfn-lint test"
+	@echo "nag	GitHub actions cfn-nag test"
+	@echo "version	[part=major||minor||patch] bump version and tag release (make version part=patch)"
+	@echo "release	push new tag to release branch"
+	@echo "clean	delete VirtualEnv and installed libraries"
+
+# Install VirtualEnv and dependencies
+init: $(VENV_NAME) pre-commit
+
+$(VENV_NAME): $(VENV_NAME)/bin/activate
+
+$(VENV_NAME)/bin/activate: requirements.txt
+	test -d $(VENV_NAME) || virtualenv -p python3 $(VENV_NAME)
+	$(PYTHON) -m pip install -U pip
+	$(PYTHON) -m pip install -Ur requirements.txt
+	touch $(VENV_NAME)/bin/activate
+
+pre-commit:
```

```suggestion
pre-commit: $(VENV_NAME)
```

the virtualenv is a dependency to running the pre-commit target, so it should be listed as such. Otherwise things could break if the virtualenv isn't initialised and someone runs just the pre-commit target or runs the init target with concurrency enabled in Make.

---

Perfect that makes sense. Thanks!